site/app/controllers/pdf/PDFController.php

```diff
@@ -87,21 +84,20 @@ public function downloadStudentPDF($gradeable_id, $filename, $path) {
         $annotation_jsons = [];
         $rerender_annotated_pdf = true;
         $latest_timestamp = filemtime(urldecode($path));
-        if (is_dir($annotation_path) && count(scandir($annotation_path)) > 2) {
-            $first_file = scandir($annotation_path)[2];
-            $annotation_path = FileUtils::joinPaths($annotation_path, $first_file);
-            if (is_file($annotation_path)) {
-                $dir_iter = new \DirectoryIterator(dirname($annotation_path . '/'));
-                foreach ($dir_iter as $fileinfo) {
-                    if (!$fileinfo->isDot()) {
-                        $no_extension = preg_replace('/\\.[^.\\s]{3,4}$/', '', $fileinfo->getFilename());
-                        $pdf_info = explode('_', $no_extension);
-                        $pdf_id = implode('_', array_slice($pdf_info, 0, -1));
-                        $grader_id = $pdf_info[count($pdf_info) - 1];
+        if (is_dir($annotation_path)) {
+            $dir_iter = new \DirectoryIterator($annotation_path);
+            foreach ($dir_iter as $file_info) {
+                if ($file_info->isFile() && !$file_info->isDot()) {
```

Comment:
When would `$annotation_path` contain anything but files?

Reply:
Rewritten to remove unneeded if statement